### PR TITLE
[19.09] matrix-synapse: 1.11.1 -> 1.12.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Backport PR for #83239 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Ma27** [comment]
@GrahamcOfBorg test matrix-synapse

**Comment by Ma27** [comment]
cc @NixOS/backports any objections?

**Review by worldofpeace — APPROVED** [review]
Yeah, keep fighting the good fight of python dependencies!!!!!
We will be victorious :joy:

File: pkgs/development/python-modules/twisted/default.nix [hunk 1]
```diff
@@ -56,5 +56,6 @@ buildPythonPackage rec {
     '';
     license = licenses.mit;
     maintainers = [ ];
+    knownVulnerabilities = [ "CVE-2020-10108" "CVE-2020-10109" ];
```
Comment (worldofpeace) [inline pkgs/development/python-modules/twisted/default.nix]: hmm, won't this mean on an upgrade any package that depends on this version of twisted will fail to evaluate?

Comment (Ma27) [reply pkgs/development/python-modules/twisted/default.nix]: Yes.

And this is basically the main reason I asked for input before merging/pushing: the thing is, to fix this package (by bumping it), it's necessary to fix several dependencies (and as you can see this affects even pytest in the end) and I'm not even sure if there's an even bigger fallout after this. So in the end, I'm absolutely not sure what's the best (acceptable) solution here :)

Comment (worldofpeace) [reply pkgs/development/python-modules/twisted/default.nix]: This is how fedora patched 10109 https://src.fedoraproject.org/rpms/python-twisted/c/9248b58fc9b22a159f50759ab4959619dfa04a04?branch=f31 you would have to do some sort of upgrade still. I don't think we can mark it as knownVulnerabilities

Comment (FRidh) [reply pkgs/development/python-modules/twisted/default.nix]: This is why it is so important we update all our core packages before doing a release. Apparently the twisted version here was already quite dated, so patching it is tough.

Comment (Ma27) [reply pkgs/development/python-modules/twisted/default.nix]: Applied the suggestion of @worldofpeace now.

**Comment by Ma27** [comment]
@NixOS/backports any further objections? :)

**Pull request merged**